lib/rules/indent.js

```diff
@@ -807,6 +889,71 @@ module.exports = {
             });
         }
 
+        /**
+        * Ignore first token of each line within an unknown node if its
+        * offset does not depend on another token's offset. If the first token
+        * of a line's offset depends on the offset of another token, make the
+        * last dependent offset depend on the first token of its line.
+        * @param {ASTNode} node Node
+        * @returns {void}
+        */
+        function ignoreUnknownNode(node) {
+            const startLine = node.loc.start.line;
+            const endLine = node.loc.end.line;
+
+            for (let lineNumber = startLine; lineNumber <= endLine; lineNumber++) {
+                const firstTokenOfLine = tokenInfo.firstTokensByLineNumber.get(lineNumber);
+
+                if (!firstTokenOfLine) {
+                    continue;
+                }
+
+                const tokenInNodeRange = (node.range[0] <= firstTokenOfLine.range[0]) &&
+                    (node.range[1] >= firstTokenOfLine.range[1]);
+
+                if (!tokenInNodeRange) {
+                    continue;
+                }
+
+                const dependency = offsets.getFirstDependency(firstTokenOfLine);
+
+                if (dependency) {
+                    let currDependency = dependency,
+                        lastDependency = null;
+
+                    while (
+                        (currDependency = offsets.getFirstDependency(currDependency)) &&
+                        currDependency.loc.start.line > startLine
+                    ) {
+                        lastDependency = currDependency;
+                    }
+
+                    if (lastDependency) {
+                        const firstTokenOfDependentLine = tokenInfo.getFirstTokenOfLine(lastDependency);
+
+                        offsets.matchIndentOf(firstTokenOfDependentLine, lastDependency);
+                    } else if (dependency.loc.start.line < startLine) {
+                        offsets.ignoreToken(firstTokenOfLine);
+                    }
+                } else {
+                    offsets.ignoreToken(firstTokenOfLine);
+                }
+            }
+        }
+
+        /**
+        * Ignore node if it is unknown
+        * @param {ASTNode} node Node
+        * @returns {void}
+        */
+        function checkForUnknownNode(node) {
+            if (KNOWN_NODES.has(node.type)) {
```

Comment:
Nitpick: It might be better to avoid the extra `return` statement here:

```js
if (!KNOWN_NODES.has(node.type)) {
    ignoreUnknownNode(node);
}
```

(In this case, I think using negation in an `if` is fine because there's no `else`.)